model/operation.cc

```diff
@@ -435,16 +435,36 @@ namespace minsky
     switch (classify(type()))
       {
       case function: case reduction: case scan: case tensor:
-        if (check && !ports[1]->units(check).empty())
-          throw_error("function input not dimensionless");
-        return {};
+        switch (type())
+          {
+          case percent:
+            { 
+              // Add % sign to units from input to % operator. Need the first conditional otherwise Minsky crashes		
+                auto r=ports[1]->units(check);	 	 
+                if (!r.empty()) {
```

Comment:
If r.empty(), you would want to return just "%". So I don't think this conditional should be here.

Reply:
When I remove this conditional I get a segfault. I have replaced it with:
if (!ports[1]->wires().empty()), which seems to work.